Applications/Spire/Source/Ui/IconButton.cpp

```diff
@@ -39,49 +45,20 @@ IconButton::IconButton(QImage icon, Style style, QWidget* parent)
     })").arg(scale_height(10)).arg(scale_height(2)).arg(scale_width(6)));
 }
 
-connection IconButton::connect_clicked_signal(
-    const ClickedSignal::slot_type& slot) const {
-  return m_clicked_signal.connect(slot);
-}
-
-void IconButton::focusInEvent(QFocusEvent* event) {
-  m_last_focus_reason = event->reason();
-  update();
-  QWidget::focusInEvent(event);
-}
-
-void IconButton::focusOutEvent(QFocusEvent* event) {
-  update();
-  QWidget::focusOutEvent(event);
-}
-
 void IconButton::keyPressEvent(QKeyEvent* event) {
-  if(event->key() == Qt::Key_Enter || event->key() == Qt::Key_Return ||
-      event->key() == Qt::Key_Space) {
-    m_clicked_signal();
-    event->accept();
-    return;
-  }
-  event->ignore();
-}
-
-void IconButton::mousePressEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton) {
-    event->accept();
-  }
-}
-
-void IconButton::mouseReleaseEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton) {
-    if(rect().contains(event->localPos().toPoint())) {
-      event->accept();
-      m_clicked_signal();
+  if(!event->isAutoRepeat()) {
+    switch(event->key()) {
+      case Qt::Key_Enter:
+      case Qt::Key_Return:
+      case Qt::Key_Space:
+        Q_EMIT clicked(isChecked());
+        return;
```

Comment:
Why was the ```event->accept()``` removed?

Reply:
QKeyEvents are accepted by default; from the [documentation](https://doc.qt.io/qt-5/qkeyevent.html) under 'Detailed Description': "A key event contains a special accept flag that indicates whether the receiver will handle the key event. This flag is set by default for QEvent::KeyPress and QEvent::KeyRelease, so there is no need to call accept() when acting on a key event."